### ISSUE TITLE
Unable to use the image with HTTPS in Azure App Service

### ISSUE THREAD

**Issue by upr-kmd** [crazy-max/docker-jetbrains-license-server#11]
I'm unable to use the image with HTTPS turned on with the container deployed to Azure App Service for Linux Containers.
The proxy container works, I can access it by web and connect it to our Jetbrains account, but I'm unable to get a license ticket from it.
It seems that JLS tries to do a reverse DNS lookup or some reverse connection on my IP and fails, because it treats it as an IPv6 address. (IP masked as XXX.YYY.ZZZ.132)

```
2019-01-24T14:18:02.425203327Z ERROR ResourceDescriptor:124 - XXX.YYY.ZZZ.132:6852: invalid IPv6 address
2019-01-24T14:18:02.425940419Z java.net.UnknownHostException: XXX.YYY.ZZZ.132:6852: invalid IPv6 address
2019-01-24T14:18:02.426293415Z 	at java.net.InetAddress.getAllByName(InetAddress.java:1169) ~[?:1.8.0_181]
2019-01-24T14:18:02.427692901Z 	at java.net.InetAddress.getAllByName(InetAddress.java:1126) ~[?:1.8.0_181]
2019-01-24T14:18:02.427863499Z 	at java.net.InetAddress.getByName(InetAddress.java:1076) ~[?:1.8.0_181]
2019-01-24T14:18:02.428019797Z 	at com.jetbrains.ls.floating.rpc.a$b.a(a$b.java:25) ~[lsrv-floating.jar:?]
2019-01-24T14:18:02.428251495Z 	at com.jetbrains.ls.floating.rpc.a.c(a.java:27) ~[lsrv-floating.jar:?]
2019-01-24T14:18:02.429154985Z 	at jetprofile.licenseService.a.b(a.java:81) ~[lsrv-floating.jar:?]
2019-01-24T14:18:02.429574381Z 	at jetprofile.licenseService.a.a(a.java:23) ~[lsrv-floating.jar:?]
2019-01-24T14:18:02.429752279Z 	at com.jetbrains.ls.floating.rpc.FloatingServerRpcHandler$a$b.a(FloatingServerRpcHandler$a$b.java:38) ~[lsrv-floating.jar:?]
2019-01-24T14:18:02.430663469Z 	at com.jetbrains.ls.floating.rpc.FloatingServerRpcHandler$a$b.invoke(FloatingServerRpcHandler$a$b.java:21) ~[lsrv-floating.jar:?]
2019-01-24T14:18:02.430851167Z 	at kara.AsyncActionResultKt$execute$1.invoke(AsyncActionResult.kt:91) ~[kara-1.0.1.jar:?]
2019-01-24T14:18:02.431008866Z 	at kara.AsyncActionResultKt$execute$1.invoke(AsyncActionResult.kt) ~[kara-1.0.1.jar:?]
2019-01-24T14:18:02.431208263Z 	at kara.ActionContextKt.withContext(ActionContext.kt:167) ~[kara-1.0.1.jar:?]
2019-01-24T14:18:02.431377662Z 	at kara.AsyncActionResultKt.execute(AsyncActionResult.kt:90) ~[kara-1.0.1.jar:?]
2019-01-24T14:18:02.432855446Z 	at kara.AsyncActionResultKt.access$execute(AsyncActionResult.kt:1) ~[kara-1.0.1.jar:?]
2019-01-24T14:18:02.433842236Z 	at kara.AsyncActionResultKt$async$1.invoke(AsyncActionResult.kt:122) [kara-1.0.1.jar:?]
2019-01-24T14:18:02.434035834Z 	at kara.AsyncActionResultKt$async$1.invoke(AsyncActionResult.kt) [kara-1.0.1.jar:?]
2019-01-24T14:18:02.435137122Z 	at kara.AsyncActionResultKt$sam$java_lang_Runnable$0.run(AsyncActionResult.kt) [kara-1.0.1.jar:?]
2019-01-24T14:18:02.435386619Z 	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) [?:1.8.0_181]
2019-01-24T14:18:02.435615417Z 	at java.util.concurrent.FutureTask.run(FutureTask.java:266) [?:1.8.0_181]
2019-01-24T14:18:02.435771615Z 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:1.8.0_181]
2019-01-24T14:18:02.435925414Z 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:1.8.0_181]
2019-01-24T14:18:02.436075912Z 	at java.lang.Thread.run(Thread.java:748) [?:1.8.0_181]
```

I've tried rebuilding the image with additional option for a reverse proxy, but it didn't help.

```
##add to entrypoint.sh
# https://www.jetbrains.com/help/license_server/configuring_secure_connection.html
if [ ! -z "$JLS_REVERSE_PROXY" ] ; then
  echo "Configuring reverse proxy..."
  license-server configure --listen-port 80 --base-url http://${JLS_REVERSE_PROXY}:443
fi
```

I've also tried enabling some Java options to enforce IPv4, but it also didn't work.

```
#add to license-server.jvmoptions.tmpl:
-Djava.net.preferIPv6Addresses=false -Djava.net.preferIPv4Stack=true
```

```
#add to Dockerfile:
COPY license-server.jvmoptions.tmpl ${JLS_PATH}/conf/license-server.jvmoptions
```

It could be 2 separate issues (not recognizing the reverse proxy, and IPv6 mixup).

**Comment by crazy-max** [comment]
@upr-kmd Have you tried an installation without Docker first to check if it's an issue with this image or not ?

**Comment by upr-kmd** [comment]
We've found the issue was with Azure's App Service proxy, which sets the following headers:
CLIENT-IP and X-Forwarded-For as IP:port, instead of just an IP, which makes JLS interpret it as an IPv6.
We've added Nginx to the docker image to fix these headers and now it works.
I've tried reporting it to Jetbrains twice, but they have ignored me (bug report to IntellIJ IDEA deleted, web contact form left without any response).
